pkg/reconciler/instances/istio/reset/pod/handler.go

```diff
@@ -35,75 +54,99 @@ type NoActionHandler struct {
 	handlerCfg
 }
 
-func (i *NoActionHandler) Execute(object CustomObject, wg GetSyncWG) {
-	defer wg().Done()
-
+func (i *NoActionHandler) ExecuteAndWaitFor(object CustomObject) error {
 	if i.debug {
-		i.log.Infof("Not doing any action for: %s %s %s", object.Kind, object.Namespace, object.Name)
+		i.log.Infof("Not doing any action for: %s/%s/%s", object.Kind, object.Namespace, object.Name)
 	}
+
+	return nil
 }
 
 type DeleteObjectHandler struct {
 	handlerCfg
 }
 
-func (i *DeleteObjectHandler) Execute(object CustomObject, wg GetSyncWG) {
-	defer wg().Done()
-
-	i.log.Infof("Deleting pod %s %s", object.Name, object.Namespace)
-	if !i.debug {
-		err := retry.Do(func() error {
-			err := i.kubeClient.CoreV1().Pods(object.Namespace).Delete(context.Background(), object.Name, metav1.DeleteOptions{})
-			if err != nil {
-				return err
-			}
-
-			i.log.Infof("Deleted pod: %s", object.Name)
-			return nil
-		}, i.retryOpts...)
+func (i *DeleteObjectHandler) ExecuteAndWaitFor(object CustomObject) error {
+	i.log.Infof("Deleting pod %s/%s", object.Namespace, object.Name)
+	if i.debug {
+		return nil
+	}
 
+	err := retry.Do(func() error {
+		err := i.kubeClient.CoreV1().Pods(object.Namespace).Delete(context.Background(), object.Name, metav1.DeleteOptions{})
 		if err != nil {
-			i.log.Error(err)
+			return err
 		}
+
+		i.log.Infof("Deleted pod %s/%s", object.Namespace, object.Name)
+		return nil
+	}, i.retryOpts...)
+
+	if err != nil {
+		return err
 	}
+
+	i.log.Infof("Not waiting for: %s/%s", object.Namespace, object.Name)
+
+	return nil
 }
 
 // RolloutHandler that restarts objects
 type RolloutHandler struct {
 	handlerCfg
 }
 
-func (i *RolloutHandler) Execute(object CustomObject, wg GetSyncWG) {
-	defer wg().Done()
-
-	i.log.Infof("Doing rollout for %s %s %s", object.Kind, object.Namespace, object.Name)
-	if !i.debug {
-		err := retry.Do(func() error {
-			err := doRollout(object, i.kubeClient)
-			if err != nil {
-				return err
-			}
-
-			i.log.Infof("Rolled out deployment: %s", object.Name)
-			return nil
-		}, i.retryOpts...)
+func (i *RolloutHandler) ExecuteAndWaitFor(object CustomObject) error {
+	i.log.Infof("Doing rollout for %s/%s/%s", object.Kind, object.Namespace, object.Name)
+	if i.debug {
+		return nil
+	}
 
+	err := retry.Do(func() error {
+		err := doRollout(object, i.kubeClient)
 		if err != nil {
-			i.log.Error(err)
+			return err
 		}
+
+		i.log.Infof("Rolled out %s/%s/%s", object.Kind, object.Namespace, object.Name)
+		return nil
+	}, i.retryOpts...)
+
+	if err != nil {
+		return err
 	}
-}
 
-type parentObject struct {
-	Name string
-	Kind string
+	return i.WaitForResources(object)
 }
 
-// CustomObject contains all necessary fields to do rollout.
-type CustomObject struct {
-	Name      string
-	Namespace string
-	Kind      string
+func (i *RolloutHandler) WaitForResources(object CustomObject) error {
```

Comment:
this is acutally the same thing the progress tracker is doing - please don't use your implementation for it.

Reply:
@lindnerby : may I ask you to also have a look on the features provided by the new progress tracker to ensure we are not missing anything. THX a lot!

Reply:
Yes, there exists a `pkg/reconciler/kubernetes/progress/ready.go` which is providing functions for ready checks. You may have a look to reuse them here.